packages/nuxt/src/core/app.ts

```diff
@@ -224,7 +224,7 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
   app.configs = []
   for (const config of layerConfigs) {
     const appConfigPath = await findPath(resolve(config.srcDir, 'app.config'))
-    if (appConfigPath) {
+    if (appConfigPath && !app.configs.includes(appConfigPath)) {
       app.configs.push(appConfigPath)
     }
```

Comment:
Middleware and plugin deduplication occur after `app:resolve`. If this should be deduplicated, perhaps it should happen there, too.

Reply:
Nice catch, we should also deduplicate after the `app:resolve` hook.

An easy way as `uniqueBy` is for array with objects would be to do:

```ts
app.configs = [...new Set(app.configs)]
```